libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -142,63 +133,59 @@ private void saveBlackList(List<String> revokedKeys) {
         Log.e(LOG_TAG, exception.getMessage());
       }
     }
+
+    saveTimestamp();
   }
 
   @Override
   public void onFailure(Call call, IOException exception) {
     Log.e(LOG_TAG, REQUEST_FAIL, exception);
+    saveTimestamp();
   }
 
   @Override
   public void onResponse(Call call, Response response) throws IOException {
-    List<String> revokedKeys = extractResponse(response);
-
-    saveBlackList(revokedKeys);
+    Log.e("test", "onResponse: " + response);
+    if (response.body() != null) {
+      saveBlackList(response.body());
+    }
   }
 
-  private String createListContent(List<String> revokedKeys) {
-    Date date = new Date();
+  private JsonObject addSHAtoKeys(JsonObject jsonObject) {
+    Gson gson = new Gson();
 
-    StringBuilder content = new StringBuilder(date.getTime() + NEW_LINE);
+    JsonArray jsonArray = jsonObject.getAsJsonArray("RevokedCertKeys");
+    JsonArray shaKeys = new JsonArray();
 
-    for (String key : revokedKeys) {
-      content.append(SHA256).append(key).append(NEW_LINE);
+    for (JsonElement key : jsonArray) {
+      shaKeys.add(SHA256 + key.getAsString());
```

Comment:
what if we send stripped version of the key to backend? also if they don't accept it should we file a ticket for them to either fix config or accept stripped version of the key?